grib2_to_cb/grib_builder_parent.py

```diff
@@ -750,6 +768,12 @@ def build_document(
                     # set the gridpoint for the station
                     station["geo"][geo_index]["x_gridpoint"] = x_gridpoint
                     station["geo"][geo_index]["y_gridpoint"] = y_gridpoint
+                # if we have gridpoints for all the geos in the station, add it to the list
+                has_gridpoints = False
+                for elem in station["geo"]:
+                    if "x_gridpoint" in elem and "y_gridpoint" in elem:
+                        has_gridpoints = True
+                if has_gridpoints:
                     self.domain_stations.append(station)
```

Comment:
This will include a station if only one of the geo elements has gridpoints. Is that what's desired here? Based on the comment, I'm wondering if this should this be something like: (I haven't tested this)

```python
# if we have gridpoints for all the geos in the station, add it to the list
station_has_all_gridpoints = all("x_gridpoint" in elem and "y_gridpoint" in elem for elem in station["geo"])
if station_has_all_gridpoints:
  self.domain_stations.append(station)
```

Reply:
I suppose it could happen where a station gets moved off of the grid for a given model and then one geo could have gridpoints where another doesn't for that station that moved away. It would still be valid, I suppose, to process an older file - in that case - where the forecastValidEpoch of the grib file was valid for the geo that was at that time located inside the grid but later is not. The risk of getting messed up statistics is pretty high, though. It might make more sense to do what you are suggesting, I think, and disallow a station completely if any of its geo entries don't have gridpoints. I think I prefer to do it like this though
```
has_gridpoints = True
for elem in station["geo"]:
    if "x_gridpoint" not in elem or  "y_gridpoint" not in elem:
        has_gridpoints = False
if has_gridpoints:
    self.domain_stations.append(station)
```
This just seems much more readable to me and it still passes my tests.